analytics/pubstack/pubstack_module_test.go

```
@@ -57,130 +55,158 @@ func loadJsonFromFile() (*analytics.AuctionObject, error) {
-	if err != nil {
-		t.Fail()
+func TestPubstackModuleErrors(t *testing.T) {
+	tests := []struct {
+		giveRefreshDelay string
+		giveMaxByteSize  string
+		giveMaxTime      string
+	}{
+		{
+			giveRefreshDelay: "1z",
+			giveMaxByteSize:  "90MB",
+			giveMaxTime:      "15m",
+		},
+		{
+			giveRefreshDelay: "1h",
+			giveMaxByteSize:  "90z",
+			giveMaxTime:      "15m",
+		},
+		{
+			giveRefreshDelay: "1z",
+			giveMaxByteSize:  "90MB",
+			giveMaxTime:      "15z",
+		},
 	}
 
-	pubstack.eventChannels[auction] = eventchannel.NewEventChannel(send, 2000, 1, 10*time.Second)
-	pubstack.eventChannels[video] = eventchannel.NewEventChannel(send, 2000, 1, 10*time.Second)
-	pubstack.eventChannels[amp] = eventchannel.NewEventChannel(send, 2000, 1, 10*time.Second)
-	pubstack.eventChannels[setUID] = eventchannel.NewEventChannel(send, 2000, 1, 10*time.Second)
-	pubstack.eventChannels[cookieSync] = eventchannel.NewEventChannel(send, 2000, 1, 10*time.Second)
-
-	pubstack.LogAuctionObject(mockedEvent)
-	pubstack.LogAmpObject(&analytics.AmpObject{
-		Status: http.StatusOK,
-	})
-	pubstack.LogCookieSyncObject(&analytics.CookieSyncObject{
-		Status: http.StatusOK,
-	})
-	pubstack.LogVideoObject(&analytics.VideoObject{
-		Status: http.StatusOK,
-	})
-	pubstack.LogSetUIDObject(&analytics.SetUIDObject{
-		Status: http.StatusOK,
-	})
-
-	pubstack.closeAllEventChannels()
-	time.Sleep(10 * time.Millisecond) // process channel
-	assert.Equal(t, counter, 0)
-
-	// Hot-Reload config
-	newFeatures := make(map[string]bool)
-	newFeatures[auction] = true
-	newFeatures[video] = true
-	newFeatures[amp] = true
-	newFeatures[cookieSync] = true
-	newFeatures[setUID] = true
-
-	remoteConfig = &Configuration{
-		ScopeID:  "new-scope",
-		Endpoint: "new-endpoint",
-		Features: newFeatures,
+	for _, tt := range tests {
+		server := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+			defer req.Body.Close()
+			data, _ := json.Marshal(&Configuration{})
```

Please never ignore errors. Consider returning a 400 status code if an error is encountered.